control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -90,14 +106,31 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
-	//TODO: Handle reconciliation.
+	// TODO: Handle reconciliation.
 
 	return ctrl.Result{}, nil
 }
 
-// SetupWithManager registers the controller with the given manager.
-func (r *GatewayController) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
+// SetupWithGatewayControllerManager registers the controller with the given manager.
+func SetupGatewayControllerWithManager(ctx context.Context, mgr ctrl.Manager, config GatewayControllerConfig) (*cache.Cache, error) {
+	c := cache.New(cache.Config{
+		ConsulClientConfig:  config.ConsulClientConfig,
+		ConsulServerConnMgr: config.ConsulServerConnMgr,
+		NamespacesEnabled:   config.NamespacesEnabled,
+		Partition:           config.Partition,
+		Kinds:               []string{api.APIGateway, api.HTTPRoute, api.TCPRoute, api.InlineCertificate},
```

Comment:
I think this would be worth pulling into a constant in the file where Cache config is defined. A user won't change this and we can keep the configuration as close to the cache as possible and in the code.

Reply:
I like that idea a lot